bchrpc/bchrpc.proto

```diff
@@ -683,7 +807,151 @@ message TransactionFilter {
     // Filter by data elements contained in pubkey scripts.
     repeated bytes data_elements = 3;
 
-    // Subscribe/Unsubscribe to everything. Other filters
-    // will be ignored.
+    // Subscribed/Unsubscribe to everything. Other filters
+    // will be ignored.    
     bool all_transactions = 4;
+
+    // Subscribed/Unsubscribe to everything SLP. Other filters
+    // will be ignored, except this filter will be overriden by all_transactions=true
+    bool all_slp_transactions = 5;
+
+    // only transactions associated with the included tokenIds
+    repeated bytes slp_token_ids = 6;
+}
+
+// SLP token info used in transaction inputs / outputs
+//
+// WARNING: Some languages (e.g., JavaScript) may not properly handle the 'uint64'
+// for large amounts. For this reason, an annotation has been added for JS to
+// return a string for the amount field instead of casting uint64 to the JS 'number'
+// type. Other languages may require similar treatment.
+//
+message SlpToken {
+    bytes token_id = 1;
+    uint64 amount = 2 [jstype = JS_STRING];
+    bool is_mint_baton = 3;
+    string address = 4;
+    uint32 decimals = 5;
+    SlpVersionType version_type = 6;
+}
+
+message SlpTransactionInfo {
```

Comment:
Should we include comments for these message types?

Reply:
We should probably add a comment for this message type.

Reply:
done.